src/main/java/frc/robot/commands/utils/NeoTrajectory.java

```diff
@@ -0,0 +1,76 @@
+package frc.robot.commands.utils;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.math.trajectory.TrajectoryConfig;
+import edu.wpi.first.math.trajectory.TrajectoryGenerator;
+import edu.wpi.first.wpilibj2.command.SwerveControllerCommand;
+import frc.robot.Constants.*;
+import frc.robot.subsystems.swerve.DriveSubsystem;
+
+import java.util.ArrayList;
+
+public class NeoTrajectory {
+    private static Translation2d lastEnd = new Translation2d(0, 0);
+    private Translation2d start;
+    private Translation2d end;
+    private double rot;
+    private TrajectoryConfig config;
+    private boolean isReversed;
+
+    public NeoTrajectory(double x, double y, double rot) {
+        this.start = lastEnd;
+        this.end = new Translation2d(x, y);
+        this.rot = rot;
+
+        isReversed = start.getX() > end.getX();
+
+        config = new TrajectoryConfig(
+            AutoConstants.kMaxSpeedMetersPerSecond,
+            AutoConstants.kMaxAccelerationMetersPerSecondSquared)
+            .setKinematics(DriveConstants.kDriveKinematics)
+            .setReversed(isReversed);
+
+        lastEnd = end; // Update the last end position
+    }
+
```

Comment:
New method addWaypoint(x, y) ->
add to poses array at position[length-1] -> Pose2d(x, y, theta)